### PR TITLE
[P2P] tracking raintree mempool deduping TODOs

### LINKED ISSUE #272: [TECHDEBT] [P2P] Optimize raintree mempool for message deduping and add tests
## Objective

Tend for the TODOes:

```go
./p2p/raintree/network.go:	// TECHDEBT(drewsky): What should we use for de-duping messages within P2P?
./p2p/raintree/network.go:	mempool map[uint64]struct{} // TODO (drewsky) replace map implementation (can grow unbounded)
./p2p/raintree/network.go:	// TODO(team): Add more tests to verify this is sufficient for deduping purposes.
```

Define scope with @Olshansk and update below

## Origin Document

[ Why? Issue justification and/or link to another document]

## Goals

- _\_*REPLACE_ME*: List of things that are high level ideas or goals driving the task_
- [ ] ...

## Deliverable

- _\_*REPLACE_ME*: List of things that are concrete deliverables_
- [ ] ...

## Non-goals / Non-deliverables

- _\_*REPLACE_ME*: List of things that are out of scope_
- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk

### PR DESCRIPTION
**Pull request by deblasis**
## Description

This PR simply links in the code the TODO items as specified in https://github.com/pokt-network/pocket/issues/272#issuecomment-1268791234

## Issue

Fixes #272

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [x] Other: issue tracking

## List of changes

- Added TODO reference

## Testing

- [ ] `make develop_test`
- [ ] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: p2p/raintree/network.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func (n *rainTreeNetwork) HandleNetworkData(data []byte) ([]byte, error) {
 
 	// Avoids this node from processing a messages / transactions is has already processed at the
 	// application layer. The logic above makes sure it is only propagated and returns.
-	// TODO(team): Add more tests to verify this is sufficient for deduping purposes.
+	// TODO (#278): Add more tests to verify this is sufficient for deduping purposes.
```
Comment (andrewnguyen22) [inline p2p/raintree/network.go:165]: Perhaps DISCUSS #278 and then TODO (issue-#) would be more appropriate

**Review by andrewnguyen22 — CHANGES_REQUESTED** [review]

File: p2p/raintree/network.go (hunk 1 above)
Comment (deblasis) [reply p2p/raintree/network.go:165]: Agreed

```suggestion
	// DISCUSS(#278): Add more tests to verify this is sufficient for deduping purposes.
```

**Review by Olshansk — APPROVED** [review]

**Comment by Olshansk** [comment]
@deblasis Can we merge this in?

**Comment by deblasis** [comment]
I think so but @andrewnguyen22 has to approve first

**Comment by Olshansk** [comment]
@deblasis These were just comment updates so I went ahead and bypassed the merge protection rules to be pragmatic about not having a block for no reason.

I took the liberty of just doing it (given the context) but lmk if you disagree and we should avoid it in the future.

**Pull request merged**